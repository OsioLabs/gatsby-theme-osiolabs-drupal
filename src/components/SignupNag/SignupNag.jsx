import React from 'react';
import { Segment } from 'semantic-ui-react';
import LoginRegisterButtonGroup from '../LoginRegisterButtonGroup/LoginRegisterButtonGroup';

const SignupNag = () => (
  <Segment.Group horizontal>
    <Segment textAlign="center">
      <p>
        Sign in with your Osio Labs account
        <br />
        to gain instant access to our entire library.
      </p>
    </Segment>
    <Segment textAlign="center">
      <LoginRegisterButtonGroup />
    </Segment>
  </Segment.Group>
);

export default SignupNag;
