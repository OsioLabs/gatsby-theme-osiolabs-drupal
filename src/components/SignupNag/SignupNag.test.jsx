import React from 'react';
import { render } from '@testing-library/react';

import SignupNag from './SignupNag';

// We don't need to test this inner component, which requires Oauth context,
// since it has it's own tests.
jest.mock('../LoginRegisterButtonGroup/LoginRegisterButtonGroup', () => () =>
  'LoginRegisterButtonGroup'
);

describe('Component: <SignupNag />', () => {
  it('renders correctly', () => {
    // Basic snapshot test.
    const { asFragment } = render(<SignupNag />);
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });
});
