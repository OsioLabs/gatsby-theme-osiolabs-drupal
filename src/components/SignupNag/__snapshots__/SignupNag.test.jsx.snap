// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <SignupNag /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ui horizontal segments"
  >
    <div
      class="ui center aligned segment"
    >
      <p>
        Sign in with your Osio Labs account
        <br />
        to gain instant access to our entire library.
      </p>
    </div>
    <div
      class="ui center aligned segment"
    >
      LoginRegisterButtonGroup
    </div>
  </div>
</DocumentFragment>
`;
